### DISCUSSION TITLE
Linux Agent

### DISCUSSION THREAD

**Discussion by lgice348** [Ylianst/MeshCentral#5020] [Q&A]
Are there any pre-reqs you should do before installing agent on a linux machine. ie ubuntu 22.04
my machines some times work and sometimes dont

**Comment by The-Real-Admin-Guy** [comment]
If you are making a reference as to the "Desktop" viewer part, it is more than likely that you need to login using Xterm instead of wayland.
This is a known issue where Wayland defaults to blocking the Remote viewers.  So on some units depending on how the user is logged in you may need to direct them to logout and login again but by clicking on the gear icon (Lower right)  and selecting login using Xterm.

You can test the agent in "Console" mode which should still work regardless of the GUI selection.  If it works then you have your solution.
I have a lot of issues where I must walk the user on allowing me in to Ubuntu since around version 18 where wayland started appearing in production system for some users.

If you need full remote and unattended access to the system you many need to disable Wayland outright and reconfigure for Xterm as default.   Otherwise you will need the user to assist, it is a security feature not actually a bug but a pain for Remote administrators....


Edit the configuration file,

sudo nano /etc/gdm3/custom.conf

to uncomment [remove # from] the line containing

WaylandEnable=false


As a side note:  You probably do not actually need access to the desktop in Ubuntu, I find the console to be amazingly functional for updates, installs and miscellaneous administrative tasks.  The desktop is only useful for training the user with applications issues and walking them through stuff.  Unlike Windows, I rarely need access to the desktop directly for anything...

**Reply by bmcgonag** [reply]
I love this response, very well put and laid out.

I'm worried, now however, that some distros are going to start removing Xorg all together, and will be using wayland only.  I hope someone will start working on the desktop viewer with wayland.  It can be done, as RustDesk has done it, and they have RDP for wayland in the default Ubuntu Gnome edition now, which works a bit differently since I think the user has to be logged in, but maybe a possiblity to use that?
